tests/common/type_list.h

```diff
@@ -170,37 +170,37 @@ inline constexpr auto get_init_value_helper<def_cnstr>(int x) {
 
 namespace get_cts_types {
 inline auto get_vector_types() {
+#if SYCL_CTS_ENABLE_OPENCL_INTEROP_TESTS
+  static const auto pack = named_type_pack<
+      bool, char, signed char, unsigned char, short, unsigned short, int,
+      unsigned int, long, unsigned long, long long, unsigned long long, float,
+      sycl::cl_float, sycl::byte, sycl::cl_bool, sycl::cl_char, sycl::cl_uchar,
+      sycl::cl_short, sycl::cl_ushort, sycl::cl_int, sycl::cl_uint,
+      sycl::cl_long, sycl::cl_ulong>::generate("bool", "char", "signed char",
+                                               "unsigned char", "short",
+                                               "unsigned short", "int",
+                                               "unsigned int", "long",
+                                               "unsigned long", "long long",
+                                               "unsigned long long", "float",
+                                               "sycl::cl_float", "sycl::byte",
+                                               "sycl::cl_bool", "sycl::cl_char",
+                                               "sycl::cl_uchar",
+                                               "sycl::cl_short",
+                                               "sycl::cl_ushort",
+                                               "sycl::cl_int", "sycl::cl_uint",
+                                               "sycl::cl_long",
+                                               "sycl::cl_ulong");
```

Comment:
I just checked, and there seem to be a number of files the use `sycl::cl_` instead of `sycl::opencl::cl_` (a quick grep for `sycl::cl_` returns 26 files). Should I just make the change from `sycl::cl_` to `sycl::opencl::cl_` in all of those files and add it to this PR?

Reply:
Yes, we should change all the other files.  It might be a little cleaner to change the namespace in a separate commit, but I don't have a strong preference.  Maybe other @KhronosGroup/sycl-cts-reviewers do?

Reply:
According to the PR title this patch is supposed to disable compilation of the test cases using OpenCL backend types unless OpenCL interop is tested explicitly. It's not related to the updating SYCL API from 1.2.1 to 2020 version.
So, I agree with @gmlueck that it is better to be done in a separate commit.